EncoreTickets.SDK/Api/Models/ApiContext.cs

```diff
@@ -34,9 +34,28 @@ public class ApiContext
 
         /// <summary>
         /// Gets or sets the affiliate.
+        /// Used as a header in requests.
         /// </summary>
         public string Affiliate { get; set; }
 
+        /// <summary>
+        /// Gets or sets the correlation ID.
+        /// Used as a header in requests.
+        /// </summary>
+        public string Correlation { get; set; }
```

Comment:
Could you please clarify how these two new properties are supposed to be used in the client code?

Reply:
The best way is to add support for external loggers in services and add logs related to a running request and use correlations within services. But before it is implemented in the SDK, it is probably worth giving a client code the opportunity to learn about correlation and use it on the client-side.

Reply:
I'm just trying to understand the way it's going to be used. I'm assuming we are going to be inheriting from the service classes and overriding the methods in order to set the correct correlation id before the request and to add appropriate logging logic. Is my understanding correct?

Reply:
I do not think that inheritance and overriding of methods are necessary. The correlation ID can simply get and set via ApiContext used for a service. And the internal logic of services already knows how to handle this from ApiContext

Reply:
ApiContext is private though so that would require keeping track of it after creation of the service or recreating the service for every request or making ApiContext public, none of which seems like a good option. Plus, we would need overriding anyway to apply logging so I guess overriding is the best option for now.